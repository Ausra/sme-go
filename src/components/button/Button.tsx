import { FunctionComponent } from "react";
import styled from "styled-components/macro";
import { defaultTheme } from "../../utils/global-styles";

export interface ButtonProps extends React.HTMLAttributes<HTMLElement> {
  dataTestId?: string;
  title?: string;
  primary?: boolean;
  margin?: string;
}

export const defaultTestId = "styled-button";

const StyledButton = styled.button<{
  primary?: boolean | undefined;
  margin?: string;
}>`
  border-radius: 24px;
  border: 1px solid ${defaultTheme.button.primaryColor};
  color: white;
  margin: ${(props) => (props.margin ? props.margin : `8px 16px`)};
  padding: 12px 24px;
  height: 42px;

  background: ${(props) =>
    props.primary
      ? defaultTheme.button.primaryColor
      : defaultTheme.button.secondaryColor};
  color: ${(props) =>
    props.primary
      ? defaultTheme.button.primaryTextColor
      : defaultTheme.button.secondaryTextColor};
`;

const Button: FunctionComponent<ButtonProps> = ({
  title,
  primary,
  onClick,
  dataTestId,
  margin,
}) => {
  return (
    <>
      <StyledButton
        primary={primary}
        onClick={onClick}
        data-testid={defaultTestId || dataTestId}
        margin={margin}
      >
        {title}
      </StyledButton>
    </>
  );
};

export default Button;
