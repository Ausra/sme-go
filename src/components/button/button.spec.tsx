import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import Button from "./Button";
import "jest-styled-components";

afterEach(cleanup);

it("button primary", () => {
  render(<Button primary />);
  const button = screen.getByRole("button");
  expect(button).toMatchSnapshot();
});
