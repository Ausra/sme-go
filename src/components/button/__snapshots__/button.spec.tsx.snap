// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button primary 1`] = `
.c0 {
  border-radius: 24px;
  border: 1px solid #2B64F5;
  color: white;
  margin: 8px 16px;
  padding: 12px 24px;
  height: 42px;
  background: #2B64F5;
  color: #ffffff;
}

<button
  class="c0"
  data-hook="styled-button"
/>
`;
